File: official/docs/java/shipments/BuyShipment.java
import com.easypost.EasyPost;
import java.util.*;

public class BuyShipment {
    public static void main(String[] args) {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");
        
        Shipment shipment = Shipment.retrieve("shp_...");

        Map<String, Object> buyMap = new HashMap<String, Object>();
        buyMap.put("rate", shipment.lowestRate());
        buyMap.put("insurance", 249.99);

        shipment.buy(buyMap);

        System.out.println(shipment);
    }
}

File: official/docs/node/shipments/buy_shipment.js
const Easypost = require('@easypost/api');
const api = new Easypost('EASYPOST_API_KEY');

api.Shipment.retrieve('shp_...').then((retrieveShipment) => {
  retrieveShipment.buy('rate_...', 249.99).then(console.log);
});
